azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/AbstractMojoBase.java

```diff
@@ -149,26 +152,29 @@ public void execute() throws MojoFailureException {
         }
     }
 
-    protected void initExecution() throws MojoFailureException, MavenDecryptException, AzureExecutionException,
-        com.microsoft.azure.toolkit.lib.auth.exception.InvalidConfigurationException {
+    protected void initExecution() throws MojoFailureException, MavenDecryptException, AzureExecutionException, LoginFailureException {
         // init proxy manager
         ProxyUtils.initProxy(Optional.ofNullable(this.session).map(s -> s.getRequest()).orElse(null));
         // Init telemetries
         initTelemetry();
+        Azure.az().config().setLogLevel(LogLevel.NONE);
+        Azure.az().config().setUserAgent(getUserAgent());
         trackMojoExecution(MojoStatus.Start);
         final MavenAuthConfiguration mavenAuthConfiguration = auth == null ? new MavenAuthConfiguration() : auth;
         mavenAuthConfiguration.setType(getAuthType());
-        this.azureCredentialWrapper = MavenAuthManager.getInstance().login(session, settingsDecrypter, mavenAuthConfiguration);
-        if (Objects.isNull(azureCredentialWrapper)) {
+        com.microsoft.azure.toolkit.lib.Azure.az(AzureAccount.class).login(
+                MavenAuthManager.getInstance().buildAuthConfiguration(session, settingsDecrypter, mavenAuthConfiguration));
+        final Account account = Azure.az(AzureAccount.class).account();
+        if (!account.isAuthenticated()) {
```

Comment:
we can simplify much of our code if we throw exception inside `account()` instead of check everywhere